code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/AEPPushTemplateNotificationBuilder.kt

```diff
@@ -0,0 +1,957 @@
+     * @param remoteView `RemoteViews` object containing a UI element to be updated
+     * @param elementId [Int] containing the resource id of the UI element
+     * @param colorHex [String] containing the color hex string
+     * @param methodName `String` containing the method to be called on the UI element to
+     * update the color
+     * @param viewFriendlyName `String` containing the friendly name of the view to be used
+     * for logging purposes
+     */
+    private fun setElementColor(
+        remoteView: RemoteViews,
+        elementId: Int,
+        colorHex: String,
+        methodName: String,
+        viewFriendlyName: String
+    ) {
+        if (StringUtils.isNullOrEmpty(methodName)) {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Null or empty method name provided, custom color will not be applied to $viewFriendlyName."
+            )
+            return
+        }
+        try {
+            if (!StringUtils.isNullOrEmpty(colorHex)) {
+                remoteView.setInt(elementId, methodName, Color.parseColor(colorHex))
+            }
+        } catch (exception: IllegalArgumentException) {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Unrecognized hex string passed to Color.parseColor(), custom color will not be applied to $viewFriendlyName."
+            )
+        }
+    }
+
+    /**
+     * Creates a click intent for the specified [Intent] action. This intent is used to handle interactions
+     * with the skip left and skip right buttons in a filmstrip or manual carousel push template notification.
+     *
+     * @param context the application [Context]
+     * @param pushTemplate the [ManualCarouselPushTemplate] object containing the manual carousel push template data
+     * @param intentAction [String] containing the intent action
+     * @param broadcastReceiverName `String` containing the broadcast receiver name
+     * @param trackerActivityName `String` containing the tracker activity name
+     * @param downloadedImageUris [List] of String` containing the downloaded image URIs
+     * @param imageCaptions `List` of String` containing the image captions
+     * @param imageClickActions `List` of String` containing the image click actions
+     * @return the created click [Intent]
+     */
+    internal fun createClickIntent(
+        context: Context,
+        pushTemplate: ManualCarouselPushTemplate,
+        intentAction: String,
+        broadcastReceiverName: String?,
+        trackerActivityName: String?,
+        downloadedImageUris: List<String?>,
+        imageCaptions: List<String?>,
+        imageClickActions: List<String?>
+    ): Intent {
+        val clickIntent = Intent(intentAction)
+        broadcastReceiverName?.let {
+            val broadcastReceiver = Class.forName(broadcastReceiverName)
+            clickIntent.setClass(context, broadcastReceiver)
```

Comment:
Can we make the `construct` entry point accept a `Class<out Activity?>` or a `Class<out BroadcastReceiver>` directly, assign it to the intent, and the let the Android system handle their creation.